src/Microsoft.AspNet.Mvc.ViewFeatures/Controller.cs

```diff
@@ -1181,6 +1183,128 @@ public virtual CreatedAtRouteResult CreatedAtRoute(string routeName, object rout
+        [NonAction]
+        public virtual ChallengeResult Challenge(IList<string> authentiationSchemes)
+            => new ChallengeResult(authentiationSchemes);
+
+        /// <summary>
+        /// Creates a <see cref="ChallengeResult"/> with the specified <paramref name="properties" />.
+        /// </summary>
+        /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the authentication
+        /// challenge.</param>
+        /// <returns>The created <see cref="ChallengeResult"/> for the response.</returns>
+        [NonAction]
+        public virtual ChallengeResult Challenge(AuthenticationProperties properties)
+            => new ChallengeResult(properties);
+
+        /// <summary>
+        /// Creates a <see cref="ChallengeResult"/> with the specified specified authentication scheme and
+        /// <paramref name="properties" />.
+        /// </summary>
+        /// <param name="authenticationScheme">The authentication scheme to challenge.</param>
+        /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the authentication
+        /// challenge.</param>
+        /// <returns>The created <see cref="ChallengeResult"/> for the response.</returns>
+        [NonAction]
+        public virtual ChallengeResult Challenge(string authentiationScheme, AuthenticationProperties properties)
+            => new ChallengeResult(authentiationScheme, properties);
+
+        /// <summary>
+        /// Creates a <see cref="ChallengeResult"/> with the specified specified authentication schemes and
+        /// <paramref name="properties" />.
+        /// </summary>
+        /// <param name="authenticationScheme">The authentication schemes to challenge.</param>
+        /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the authentication
+        /// challenge.</param>
+        /// <returns>The created <see cref="ChallengeResult"/> for the response.</returns>
+        [NonAction]
+        public virtual ChallengeResult Challenge(
+            IList<string> authentiationSchemes,
+            AuthenticationProperties properties)
+            => new ChallengeResult(authentiationSchemes, properties);
+
+        /// <summary>
+        /// Creates a <see cref="ForbiddenResult"/>.
+        /// </summary>
+        /// <returns>The created <see cref="ForbiddenResult"/> for the response.</returns>
+        [NonAction]
+        public virtual ForbiddenResult Forbid()
+            => new ForbiddenResult();
+
+        /// <summary>
+        /// Creates a <see cref="ForbiddenResult"/> with the specified authentication scheme.
+        /// </summary>
+        /// <param name="authenticationScheme">The authentication scheme to challenge.</param>
+        /// <returns>The created <see cref="ForbiddenResult"/> for the response.</returns>
+        [NonAction]
+        public virtual ForbiddenResult Forbid(string authentiationScheme)
```

Comment:
Why not `Forbidden`? I guess it sounds weird either way... but do we ever have much of a deviation between the factory name and the result type?